angular/src/app/shared/recommendations-service/recommendations.service.spec.ts

```diff
@@ -274,5 +288,23 @@ describe('RecommendationsService', () => {
                 expect(cavityWallInsulationRecommendation.grant.name).toBe('National Grant For Measure');
             });
         }));
+
+        it('should save recommendations to session', async(() => {
+            // when
+            const recommendationsObservable = service.getAllRecommendations(dummyEnergyCalculations);
+
+            // then
+            recommendationsObservable.toPromise().then((recommendations) => {
```

Comment:
does this kind of async test work in this form? I'd imagine it ends without resolving the `then`?
I think the options here are to have the test "expect" a certain number of assertions, or (probably easier) I think you can just await the promise and then it'll _just work_

Reply:
This seems to be done multiple times in file/ project, and the expects are called correctly (changing the expected measures will cause the text to fail).

So I think it is fine. Unless I have misunderstood? Let me know if so and we can discuss further.

Reply:
eh, if it works then I'm happy- I must be misremembering from a different project. Cheers!